src/main/api/2/studio-api-2.yaml

```diff
@@ -1456,6 +1460,235 @@ paths:
+        '401':
+          $ref: '#/components/responses/Unauthorized'
+        '404':
+          $ref: '#/components/responses/NotFound'
+        '409':
+          $ref: '#/components/responses/Conflict'
+        '500':
+          $ref: '#/components/responses/InternalServerError'          
+
+  /repository/pull_from_remote:
+    post:
+      tags:
+        - repository
+      summary: Pull content from remote repository to site content repository
+      operationId: pullFromRemoteRepository
+      requestBody:
+        description: pull from remote repository request body
+        required: true
+        content:
+          application/json:
+            schema:
+              type: object
+              properties:
+                siteId:
+                  type: string
+                  description: Site ID
+                remoteName:
+                  type: string
+                  description: Remote repository name to pull from
+                remoteBranch:
+                  type: string
+                  description: Remote repository branch to pull from
+      responses:
+        '200':
+          description: OK
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  response:
+                    $ref: '#/components/schemas/ApiResponse'
+        '400':
+          $ref: '#/components/responses/BadRequest'
+        '401':
+          $ref: '#/components/responses/Unauthorized'
+        '404':
+          $ref: '#/components/responses/NotFound'
+        '500':
+          $ref: '#/components/responses/InternalServerError'
+
+  /repository/push_to_remote:
+    post:
+      tags:
+        - repository
+      summary: Push content to remote repository from site content repository
+      operationId: pushToRemoteRepository
+      requestBody:
+        description: push to remote repository request body
+        required: true
+        content:
+          application/json:
+            schema:
+              type: object
+              properties:
+                siteId:
+                  type: string
+                  description: Site ID
+                remoteName:
+                  type: string
+                  description: Remote repository name to push to
+                remoteBranch:
+                  type: string
+                  description: Remote repository branch to push to
+      responses:
+        '200':
+          description: OK
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  response:
+                    $ref: '#/components/schemas/ApiResponse'
+        '400':
+          $ref: '#/components/responses/BadRequest'
+        '401':
+          $ref: '#/components/responses/Unauthorized'
+        '404':
+          $ref: '#/components/responses/NotFound'
+        '500':
```

Comment:
Add `409` and indicate git conflict and the user should notify the administrator.